spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/common/HazelcastIntegrationDefinitionValidator.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.common;
+
+import java.util.Set;
+
+import reactor.util.StringUtils;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.IList;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.IQueue;
+import com.hazelcast.core.ISet;
+
+/**
+ * This is common validator class for SI-Hazelcast Support. It validates cache events and
+ * types.
+ * 
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ *
+ */
+public class HazelcastIntegrationDefinitionValidator {
+
+	public static <E extends Enum<E>> boolean validateEnumType(Class<E> enumType, String cacheEventTypes) {
+		Set<String> eventTypeSet = StringUtils.commaDelimitedListToSet(cacheEventTypes);
+		for (String eventType : eventTypeSet) {
+			Enum.valueOf(enumType, eventType);
+		}
+
+		return true;
+	}
+
+	public static boolean validateCacheType(DistributedObject distributedObject) {
+		if (distributedObject instanceof IMap 
+				|| distributedObject instanceof IList
+				|| distributedObject instanceof ISet
+				|| distributedObject instanceof IQueue) {
+			return true;
+		}
+
+		return false;
+	}
+
+	public static boolean validateCacheEventByDistributedObject(
+			DistributedObject distributedObject, Set<String> cacheEventTypeSet) {
+		if ((distributedObject instanceof IList) 
+				|| (distributedObject instanceof ISet)
+				|| (distributedObject instanceof IQueue)) {
+
+			for (String cacheEventType : cacheEventTypeSet) {
+				if (!(CacheEventType.ADDED.toString().equals(cacheEventType) 
+						|| CacheEventType.REMOVED.toString().equals(cacheEventType))) {
+					throw new IllegalStateException(
+							"'cache-events' attribute of IList, ISet or IQueue can be set as only "
+									+ CacheEventType.ADDED.toString() + " and / or "
+									+ CacheEventType.REMOVED.toString());
+				}
+			}
+
+		}
+
+		return true;
```

Comment:
**MultiMap** and **ReplicatedMap** should be validated as well for unsupported cache types. Supported cache event types are as follows :

1. Supported cache event types for IMap : ADDED, REMOVED, UPDATED, EVICTED, EVICT_ALL and CLEAR_ALL. 
2. Supported cache event types for MultiMap : ADDED, REMOVED and CLEAR_ALL.
3. Supported cache event types for ReplicatedMap : ADDED, REMOVED, UPDATED, EVICTED. 
4. Supported cache event types for IList, ISet and IQueue : ADDED, REMOVED. 
5. There is no need to cache event type definition for ITopic.